Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -41,6 +44,90 @@ struct IsOutsideDomainBoundary {
     }
 };
 
+struct FindBoundaryIntersection {
```

Comment:
Actually: could we rename this to `FindEmbeddedBoundaryIntersection`, to be more explicit.